ibis/tests/all/test_speedcheck.py

```diff
@@ -0,0 +1,26 @@
+from datetime import datetime
+
+import ibis
+import ibis.bigquery as bq
+import ibis.expr.operations as ops
+from ibis.expr.types import TableExpr
+
+
+def test_large_compile():
+    num_columns = 20
+    num_joins = 7
+
+    names = [f"col_{i}" for i in range(num_columns)]
+    schema = ibis.Schema(names, ['string'] * num_columns)
+    ibis_client = bq.BigQueryClient.__new__(bq.BigQueryClient)
```

Comment:
I think you can make this a little simpler

```
In [35]: table = TableExpr(ops.SQLQueryResult("select * from t", schema, SQLClient()))                                                                                                                      

In [37]: for _ in range(num_joins): 
    ...:     table = table.mutate(dummy=ibis.literal("")) 
    ...:     table = table.left_join(table, ["dummy"])[[table]] 
    ...:                                                                                                                                                                                                    

In [38]: impala.compile(table)        
```

Reply:
Nvm, I think we do want to use the bigquery compiler here. Let me see how to use that easily.

Reply:
The use of `__new__` is to skip having to authenticate

Reply:
OK, I think this is the easiest:

```
In [42]: class MockBigQueryClient(BigQueryClient): 
    ...:     def __init__(self): 
    ...:         pass 
    ...:                                                                                                                                                                                                    

In [43]: client = MockBigQueryClient()                                                                                                                                                                      

In [44]: table = TableExpr(ops.SQLQueryResult("select * from t", schema, client))                                                                                                                           

In [45]: table.compile()                                                                                                                                                                                    
Out[45]: 'SELECT *\nFROM (\n  select * from t\n) t0'

In [46]: for _ in range(num_joins): 
    ...:     table = table.mutate(dummy=ibis.literal("")) 
    ...:     table = table.left_join(table, ["dummy"])[[table]] 
    ...:                                                                                                                                                                                                    

In [47]: table.compile()